src/test/java/io/cdap/plugin/topn/TopNConfigTest.java

```diff
@@ -0,0 +1,136 @@
+package io.cdap.plugin.topn;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.validation.CauseAttributes;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.cdap.etl.api.validation.ValidationFailure;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class TopNConfigTest {
+
+  private static final Schema SCHEMA =
+    Schema.recordOf("people",
+                    Schema.Field.of("name", Schema.of(Schema.Type.STRING)),
+                    Schema.Field.of("id", Schema.of(Schema.Type.LONG)),
+                    Schema.Field.of("kg", Schema.of(Schema.Type.DOUBLE)),
+                    Schema.Field.of("cm", Schema.of(Schema.Type.FLOAT)),
+                    Schema.Field.of("age", Schema.nullableOf(Schema.of(Schema.Type.INT))));
+  private static final String STAGE = "stage";
+  private static final String NON_EXIST_FIELD = "nonExist";
+  private static final String NON_NUMERIC_FIELD = "name";
+  private static final String CORRECT_FIELD = "id";
+
+  @Test
+  public void testValidateFailOnNonExistField() {
+    TopNConfig config = new TopNConfig(NON_EXIST_FIELD, 4, false);
+    MockFailureCollector failureCollector = new MockFailureCollector(STAGE);
+    try {
+      config.validate(failureCollector, SCHEMA);
+    } catch (ValidationException e) {
+      assertEquals(1, e.getFailures().size());
+      ValidationFailure validationFailure = e.getFailures().get(0);
+      assertEquals(String.format("Field '%s' you have specified does not exist in the input.", NON_EXIST_FIELD),
+                   validationFailure.getMessage());
+      assertNull(validationFailure.getCorrectiveAction());
+      assertEquals(1, validationFailure.getCauses().size());
+      ValidationFailure.Cause cause = validationFailure.getCauses().get(0);
+      assertEquals(TopNConfig.FIELD, cause.getAttribute(CauseAttributes.STAGE_CONFIG));
+      assertEquals(STAGE, cause.getAttribute("stage"));
+    }
+  }
+
+  @Test
+  public void testValidateFailOnNonNumericField() {
+    TopNConfig config = new TopNConfig(NON_NUMERIC_FIELD, 4, false);
+    MockFailureCollector failureCollector = new MockFailureCollector(STAGE);
+    try {
+      config.validate(failureCollector, SCHEMA);
+    } catch (ValidationException e) {
+      assertEquals(1, e.getFailures().size());
+      ValidationFailure validationFailure = e.getFailures().get(0);
+      assertEquals(String.format(
+        "Field '%s' is of type '%s', only types allowed are numeric types.", NON_NUMERIC_FIELD,
```

Comment:
We should not check exception messages, just check failed property and stacktrace presence.